state/block.go

```diff
@@ -113,6 +117,8 @@ func (bs *blockState) GetHeader(hash common.Hash) (*types.Header, error) {
 
 // GetBlockData returns a BlockData for a given hash
 func (bs *blockState) GetBlockData(hash common.Hash) (types.BlockData, error) {
+	//bs.lock.Lock()
```

Comment:
I think it would be preferred to have the locks in the state service rather than the database service, since the database will only be accessed through the state service. is there a reason these are commented out?

Reply:
I deleted it few mins before your comment but did not push, so its gone now. 
--
now do it in the service level its possible as well, I will check how it works...